README.md

```diff
@@ -332,6 +332,22 @@ curl http://127.0.0.1:8080/such_path
 > such_path
 ```
 
+## Migrating to 2.x from 1.x
+
+The main change is the update to `path-to-regexp@2.0.0`, which has a few breaking changes:
+
+- No longer a direct conversion to a RegExp with sugar on top - it's a path matcher with named and unnamed matching groups
+  - It's unlikely you previously abused this feature, it's rare and you could always use a RegExp instead
```

Comment:
We should explain how to know if you are using the feature rather than editorial that "you're probably not using it"

Reply:
@blakeembrey I copied this from your readme.  Can you give me a short blurb to help explain what this means?

Reply:
Sure. It's anywhere someone is abusing stuff like `/\\d+` outside of a matching group (e.g. instead of `/(\\d+)`). It's possible it's used in places today, but wouldn't be too useful since it wouldn't result in a match the user could use.

Reply:
If you end up coming up with a better or more succinct explanation, happy to replace the one in the README 😄

Reply:
You'd be surprised what people do with these routes. Express 4.16.1 is a release just because it seemed to be popular to do something thag shouldn't work, but did.

Reply:
Question: What specifically does this mean?

> Parameters have suffixes that augment meaning - `*`, `+` and `?`. E.g. `/:user*`

Couldn't we always use those suffixes?  How are they different from `0.1.7` to `2.0.0`?

Reply:
No, 0.x never had a concept of "suffixes". There's places where it happened to work because everything is treated as a regexp with based search and replace. E.g. `/*` became `/(.*)` in search and replace (that's the reason you can't use `*` in Express.js). It also means that, yes, stuff like `/:user?` might have worked but purely because of the search and replace implications and definitely not consistently (for instance, `/:test*` never would have done anything expected).

Note: I haven't tested the specific examples above, but it's possible a comparison (0.x -> 2.x) might be the easiest way to understand the differences.

Reply:
Yeah that is probably why those worked for me :)

I was actually thinking it might be nice to update the demo app [1] to support multiple versions, which might be helpful for people to discover the differences for their routes.  Maybe I can work on that next.

[1]: http://forbeslindesay.github.io/express-route-tester/

Reply:
Ok, got a PR for that feature in the demo app: https://github.com/ForbesLindesay/express-route-tester/pull/11

Reply:
If you all need to see the differences in the versions you can view it here until the PR is merged: https://wesleytodd.github.io/express-route-tester/